google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java

```diff
@@ -0,0 +1,197 @@
+import java.util.Set;
+
+import javax.swing.JComponent;
+import javax.swing.JPanel;
+
+/**
+ * @author nik
+ */
+public class AppEngineStandardFacetEditor extends FacetEditorTab {
+
+  private final AppEngineFacetConfiguration facetConfiguration;
+  private final FacetEditorContext context;
+  private JPanel myMainPanel;
+  private AppEngineStandardLibraryPanel appEngineStandardLibraryPanel;
+  private Listener libraryListener;
+
+
+  public AppEngineStandardFacetEditor(AppEngineFacetConfiguration facetConfiguration,
+      FacetEditorContext context) {
+    this.facetConfiguration = facetConfiguration;
+    this.context = context;
+    libraryListener = new LibraryModificationListener();
+
+    LibraryTablesRegistrar.getInstance()
+        .getLibraryTable(context.getProject()).addListener(libraryListener);
+  }
+
+  @Nls
+  public String getDisplayName() {
+    return "Google App Engine";
+  }
+
+  @NotNull
+  public JComponent createComponent() {
+    return myMainPanel;
+  }
+
+  @Override
+  public boolean isModified() {
+    Set<AppEngineStandardMavenLibrary> savedLibs
+        = facetConfiguration.getLibraries(context.getProject());
+    Set<AppEngineStandardMavenLibrary> selectedLibs
+        = appEngineStandardLibraryPanel.getSelectedLibraries();
+
+    return !Objects.equals(savedLibs, selectedLibs);
+  }
+
+  @Override
+  public void apply() {
+    Set<AppEngineStandardMavenLibrary> savedLibs
+        = facetConfiguration.getLibraries(context.getProject());
+    Set<AppEngineStandardMavenLibrary> selectedLibs
+        = appEngineStandardLibraryPanel.getSelectedLibraries();
+
+    Set<AppEngineStandardMavenLibrary> libsToAdd = Sets.difference(selectedLibs, savedLibs);
+    Set<AppEngineStandardMavenLibrary> libsToRemove = Sets.difference(savedLibs, selectedLibs);
+
+    if (!libsToAdd.isEmpty()) {
+      final ModifiableRootModel rootModel
+          = ModuleRootManager.getInstance(context.getModule()).getModifiableModel();
+      AppEngineSupportProvider.addMavenLibraries(libsToAdd, context.getModule(),
+          rootModel,
+          AppEngineSupportProvider
+              .findOrCreateWebArtifact((AppEngineFacet) context.getFacet()));
+    }
+
+    if (!libsToRemove.isEmpty()) {
+      AppEngineSupportProvider.removeMavenLibraries(libsToRemove, context.getModule());
+    }
+  }
+
+  @Override
+  public void reset() {
+    appEngineStandardLibraryPanel
+        .selectLibraries(facetConfiguration.getLibraries(context.getProject()));
+  }
+
+  @SuppressWarnings("checkstyle:abbreviationaswordinname")
+  public void disposeUIResources() {
+    if (libraryListener != null) {
+      LibraryTablesRegistrar.getInstance()
+          .getLibraryTable(context.getProject()).removeListener(libraryListener);
+    }
+  }
+
+  @Override
+  public String getHelpTopic() {
+    return "Google_App_Engine_Facet";
+  }
+
+  @Override
+  public void onFacetInitialized(@NotNull Facet facet) {
+    AppEngineWebIntegration.getInstance().setupDevServer();
```

Comment:
this is not new code (I'm not sure why github didn't catch the file renaming).

I believe this is for when you create a new app engine standard project and you have a local dev server set up by default for you.